cpp/src/arrow/flight/sql/odbc/flight_sql/CMakeLists.txt

```diff
@@ -0,0 +1,176 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+cmake_minimum_required(VERSION 3.16)
+set(CMAKE_CXX_STANDARD 17)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+include_directories(include include/flight_sql
+                    ${CMAKE_SOURCE_DIR}/odbcabstraction/include)
+
+if(DEFINED CMAKE_TOOLCHAIN_FILE)
+  include(${CMAKE_TOOLCHAIN_FILE})
+endif()
+
+find_package(RapidJSON CONFIG REQUIRED)
+
+if(MSVC)
+  # the following definitions stop arrow from using __declspec when statically
+  # linking and will break on Windows without them
+  add_compile_definitions(ARROW_STATIC ARROW_FLIGHT_STATIC)
+endif()
```

Comment:
We don't need this because `arrow_static`/`arrow_flight_static` CMake targets do this automatically.

Reply:
removed